Yank/restraints.py

```diff
@@ -1992,6 +2048,297 @@ def _assign_if_undefined(attr_name, attr_value):
+        system = thermodynamic_state.system
+        self._add_force_in_separate_group(system, restraint_force)
+        thermodynamic_state.system = system
+
+    def get_standard_state_correction(self, thermodynamic_state):
+        """Return the standard state correction.
+
+        Parameters
+        ----------
+        thermodynamic_state : openmmtools.states.ThermodynamicState
+            The thermodynamic state.
+
+        Returns
+        -------
+        DeltaG : float
+           Computed standard-state correction in dimensionless units (kT).
+
+        """
+        return self._get_standard_state_correction_numerical(thermodynamic_state)
+        # TODO: Is this required, or is there only one way to do it?
+        # if self.standard_state_correction_method == 'analytical' and False:
+        #     # This code should not be reachable on purpose until we figure out what the SSC is
+        #     return self._get_standard_state_correction_analytical(thermodynamic_state)
+        # else:  # The property checks that the value is known in the setter.
+        #     return self._get_standard_state_correction_numerical(thermodynamic_state)
+
+    def determine_missing_parameters(self, thermodynamic_state, sampler_state, topography):
+        """Set reference positions for RMSD restraint.
+
+        Future iterations of this feature will introduce the ability to extract
+        equilibrium parameters and spring constants from a short simulation.
+
+        Parameters
+        ----------
+        thermodynamic_state : openmmtools.states.ThermodynamicState
+            The thermodynamic state.
+        sampler_state : openmmtools.states.SamplerState, optional
+            The sampler state holding the positions of all atoms to be used as reference
+        topography : yank.Topography, optional
+            The topography with labeled receptor and ligand atoms.
+
+        """
+
+        # Pick out the restrained atoms
+        if not self._are_restrained_atoms_defined:
+            self._pick_restrained_atoms(topography)
+
+        # We determine automatically only the parameters that have been left undefined.
+        def _assign_if_undefined(attr_name, attr_value):
+            """Assign value to self.name only if it is None."""
+            if getattr(self, attr_name) is None:
+                setattr(self, attr_name, attr_value)
+
+        # Set spring constants uniformly, as in Ref [1] Table 1 caption.
+        _assign_if_undefined('K_RMSD', 0.6 * unit.kilocalories_per_mole / unit.angstrom**2)
+        _assign_if_undefined('RMSD0', 2.0 * unit.angstroms)
+
+        # Write restraint parameters
+        msg = 'restraint parameters:\n'
+        for parameter_name, parameter_value in self._parameters.items():
+            msg += '{0:24s} : {1}\n'.format(parameter_name, parameter_value)
+        logger.debug(msg)
+
+        # Assign the sampler state
+        _assign_if_undefined('reference_state', sampler_state)
```

Comment:
@andrrizzi I'd like your opinion here.

This block reassigns the `reference_state` if it has not been defined. However, that assumes you do not want to overwrite the `reference_state` from `__init__`. Should that be the default behavior, or should `determine_missing_parameters` overwrite `__init__` parameters? This is the only time it has happened where the parameter has a 1-to-1 match in `__init__` and `determine_missing_parameters`.

My thought is that `__init__` is priority and `determine_missing_parameters` should only fill in what was not already provided.

Reply:
> My thought is that __init__ is priority and determine_missing_parameters should only fill in what was not already provided.

I agree!